### DISCUSSION TITLE
Is it ok to run full nodes using a same keys on many machines?

### DISCUSSION THREAD

**Discussion by pluto2021** [Chia-Network/chia-blockchain#2284] [Q&A]
Is it OK to run the main chia node on multiple machines? (understanding that there would be multiple chains stored). Just want to know if it would generate any errors.

**Comment by fabitom** [comment]
[Farming-on-many-machines](https://github.com/Chia-Network/chia-blockchain/wiki/Farming-on-many-machines)
````Each harvester must have a different set of TLS certificates for your main machine to recognize it as different harvesters. Unintended bugs can occur, including harvesters failing to work properly when the same certificates are shared among different machines.````

**Reply by costi** [reply]
I think the question was about combo full node, farmer and harvester on different machines. As far as I know there could be an issue if they're on the same network and they take the same incoming port using UPNP. I would like some confirmation that there's no issue otherwise.

**Reply by pluto2021** [reply]
If say keeping two main nodes using a same key on different network combos as below, which pairs would be working fine?

(1) Wifi from 2 different ISPs, say one from home network provider, the other from cellular network provider.

(2) Both connected to a same Wifi hot spot

(3) One hardwired thru Ethernet cable and the other thru wifi on a same router.

Main reason is that the command line setup for the multi harvester is cumbersome and not very reliable when you have wifi signal dropping issues from time to time. 

Hope that a GUI is developed to handle all these scenarios.
